prowler/providers/aws/services/sqs/sqs_queues_not_publicly_accessible/sqs_queues_not_publicly_accessible.py

```diff
@@ -28,14 +31,18 @@ def execute(self):
                                 and "*" in statement["Principal"]["CanonicalUser"]
                             )
                         ):
-                            if "Condition" not in statement:
+                            if (
+                                "Condition" in statement
+                                and is_account_only_allowed_in_condition(
+                                    statement["Condition"], sqs_client.audited_account
+                                )
+                            ):
+                                report.status_extended = f"SQS queue {queue.id} is not public due to filter in queue resource policy condition block"
```

Comment:
```suggestion
                                report.status_extended = f"SQS queue {queue.id} is not public because their policy only allows access from the same account"```